spinnman/processes/application_copy_run_process.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) 2017-2019 The University of Manchester
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from spinnman.messages.scp.impl import AppCopyRun
+from .abstract_multi_connection_process import AbstractMultiConnectionProcess
+
+
+def _get_next_chips(old_next_chips, machine, chips_done):
+    next_chips = set()
+    for _old_link, chip in old_next_chips:
+        for link in chip.router.links:
+            chip_coords = (link.destination_x, link.destination_y)
+            if chip_coords not in chips_done:
+                next_chip = machine.get_chip_at(*chip_coords)
+                if not next_chip.virtual:
+                    opp_link = (link.source_link_id + 3) % 6
+                    next_chips.add((opp_link, next_chip))
+    return next_chips
+
+
+class ApplicationCopyRunProcess(AbstractMultiConnectionProcess):
```

Comment:
*Technically* this ought to be documented...

Reply:
Docs added